lib/Dialect/FIRRTL/Transforms/BlackBoxReader.cpp

```diff
@@ -260,10 +238,23 @@ void BlackBoxReaderPass::runOnOperation() {
 
       LLVM_DEBUG(annotationInfo.print(llvm::dbgs().indent(6) << "- ", 8));
 
-      auto &bestAnnotationInfo = emittedFileMap[annotationInfo.name];
-      if (annotationInfo.priority < bestAnnotationInfo.priority) {
-        bestAnnotationInfo = annotationInfo;
-
+      // If we have seen a black box trying to create a blackbox with this
+      // filename before, then compute the lowest commmon ancestor between the
+      // two blackbox paths.  This is the same logic used in `AssignOutputDirs`.
+      // However, this needs to incorporate filenames that are only available
+      // _after_ output directories are assigned.
+      auto *ptr = emittedFileMap.find(annotationInfo.name);
```

Comment:
nit: doing a find followed by insert-if-not-there can be collapsed into something like `try_emplace`?
Probably not a big deal here but good habit to get into avoiding the double-lookup (hash or tree walk or however that's done).

Reply:
:+1: Fixed in: https://github.com/llvm/circt/pull/7541/commits/61a594cbcd44eed7df834d0f9d3dc86bcaf62486